src/main/java/org/ballerinalang/command/cmd/ListCommand.java

```diff
@@ -91,31 +98,42 @@ public void setParentCmdParser(CommandLine parentCmdParser) {
      * @param outStream stream outputs need to be printed
      */
     private static void listDistributions(PrintStream outStream) {
+        String currentBallerinaVersion = ToolUtil.getCurrentBallerinaVersion();
         try {
-            String currentBallerinaVersion = ToolUtil.getCurrentBallerinaVersion();
             File folder = new File(ToolUtil.getDistributionsPath());
             File[] listOfFiles;
             listOfFiles = folder.listFiles();
+            Arrays.sort(listOfFiles);
+            List<Channel> channels = ToolUtil.getDistributions();
             outStream.println("Distributions available locally: \n");
             List<String> installedVersions = new ArrayList<>();
-            for (int i = 0; i < listOfFiles.length; i++) {
-                if (listOfFiles[i].isDirectory()) {
-                    String version = listOfFiles[i].getName().split("-")[1];
-                    outStream.println(markVersion(currentBallerinaVersion, version)
-                            + " " +  ToolUtil.getType(version) + " version " + version);
-                    installedVersions.add(version);
+            for (Channel channel : channels) {
+                for (Distribution distribution : channel.getDistributions()) {
+                    for (int i = 0; i < listOfFiles.length; i++) {
+                        if (listOfFiles[i].isDirectory()) {
+                            String version = listOfFiles[i].getName().split("-")[1];
+                            if (version.equals(distribution.getVersion())) {
+                                outStream.println(markVersion(currentBallerinaVersion, version)
+                                        + " " + ToolUtil.getTypeName(version));
+                                installedVersions.add(version);
+                            }
+                        }
+                    }
                 }
             }
+            writeLocalDistsIntoJson(installedVersions);
             outStream.println("\nDistributions available remotely:");
-            List<Channel> channels = ToolUtil.getDistributions();
             for (Channel channel : channels) {
                 outStream.println("\n" + channel.getName() + "\n");
                 for (Distribution distribution : channel.getDistributions()) {
-                    outStream.println("  [" + distribution.getVersion() + "] " + distribution.getName());
+                    outStream.println(markVersion(currentBallerinaVersion, distribution.getVersion())
+                            + " " + distribution.getName());
                 }
             }
-        } catch (CommandException e) {
-            ErrorUtil.printLauncherException(e, outStream);
+        } catch (CommandException | IOException e) {
+            ErrorUtil.printLauncherException((CommandException) e, outStream);
```

Comment:
@parkavi11 Here are we printing exception in the output if there is no internet?
Also here e could IOException as well and we are casting it to CommandException